### ISSUE TITLE
Remove closing comment syntax

### ISSUE THREAD

**Issue by chrisgrieser** [maegul/comment_marks#3]
Thanks for this plugin! This is actually something I have been looking for for a while now, since Table of Comments wasn't usable in bigger files. This plugin, however, still performances well in big files, so I can finally use it :)

I use it to navigate a css file, and with the CSS comment Syntax, it displays the closing `*/`, which unnecessarily clutter the plugin's goto. Could those be maybe removed?
<img width="353" alt="Screenshot 2022-02-07 15 33 29" src="https://user-images.githubusercontent.com/73286100/152808199-9b3967cf-a08e-4fdf-b393-cdc78648afad.png">

**Comment by maegul** [comment]
Good to hear the performance on larger files is holding up!

To get it any better would probably require making custom syntax definitions to take advantage of Sublime's syntax analysis and tokenisation.  Which is probably doable but a bit more hacky.

---

Thanks for this report!  It's a problem that wouldn't just affect CSS but any trailing comment syntax (eg HTML), so it's a bug and should be fixed.

Though, trailing comment syntaxes allow for comments to be multi-line within single comment symbols/tokens.  So a lazy response would be `just put the */ on another line`.  Which you could do, but that's not really satisfactory I think

Should definitely be possible to fix, but not super, super trivially, it will probably require another setting, and, the multiline thing might cause some problems (?), so don't expect a fix to be really quick.  

---

Off the top of my head, FYI, the best way to do it would be to apply a regex to the `Comment Marks` before they're displayed in the quickfind panel, capturing everything before the trailing comment.

Alternatively, the regex for searching for `Comment Marks` could include the trailing comment and simply not capture it, which would probably be faster but might run into some bugs with multiline comments.

Either way, exposing the regex in the settings is probably a must, as you never know what strange syntax is out there!

**Comment by maegul** [comment]
For instance, the current default regex for finding `Comment Marks` is:

```regex
#                                                                                 |-- Your heading
#                                                                                 |
# ------------     alternative comment start characters -----------  | ">"levels  V
(^[ \t]*\#+|^[ \t]*/+|^[ \t]*/\*+|^[ \t]*%+|^[ \t]*<!\-\-+|^[ \t]*\-+)[ ]*(>+)\s*(.+)
```

FYI, this gets "compiled" at startup and printed to the console.

Picking up trailing comment symbols could be done like so ... with alternative matches for trailing comments characters (including none at all) which could be picked up from a `scope_end_comment_chars` setting.

```regex
#                                                                                 |-- Your heading
#                                                                                 |
# ------------     alternative comment start characters -----------  | ">"levels  V  | -trailing- |
(^[ \t]*\#+|^[ \t]*/+|^[ \t]*/\*+|^[ \t]*%+|^[ \t]*<!\-\-+|^[ \t]*\-+)[ ]*(>+)\s*(.+)(\*/$|\-\->$|)
```

**Comment by chrisgrieser** [comment]
I am not totally sure I can follow everything, but wouldn't it be possible to solve this much easier by simply stripping the trailing comment syntax *only from the dropdown menu*?

I mean, the trailing comment syntax is only an annoyance visually, so just removing it from the displayed results only should suffice. And apart from being a simple solution, it would probably also be better on performance, since you don't need additional regexes? :P

**Comment by maegul** [comment]
Yep agreed. I’m pretty sure I said the same thing somewhere in my ramblings above 😄. I’m just a smidge concerned whether that might hit performance (however unlikely).

**Comment by chrisgrieser** [comment]
well, I have a css file that's almost 2mb big, and at least on my end, the plugin works almost instantly. So I am optimistic in this regard :)

**Comment by maegul** [comment]
Just pushed up a new version with a fix for this (`v 0.1.11`).  It can take a bit of time (~ 30 mins?) for Package Control to pick up new versions though.

**NOTE** that there is a configuration parameter to turn the fix on and off and that it is (for now) off by default.  The setting is `trim_trailing_comment_chars`, which you'll need to set to `true` to have trailing comment characters/delimiters trimmed from the command palette.

The reason I've set it to `false` by default is that there's a chance that the regex the plugin is using has some problem I'm not aware of.  It was natural to me, but there's a chance some deeper regex/finite state machine knowledge I lack might reveal an issue.  Happy to take feedback of course!

Nonetheless it appears to work well for me, so switch it on and let me know if there are any problems!!

---

FYI, the essential regex is, for `CSS`: `(.*?)( *\*+/ *$|$)`, where `*/` is being captured in the second group.

The seemingly *tricky* bit is the combination of the lazy `*?` in the first group and the blank until the end of the line alternative in the second group (ie, `...|$)`).  

The idea is to capture everything until the end of the line (with the `.*` in the first group, ending with the plain `$` in the second group) ... unless there's a `*/` at the end of the line,  in which case capture everything up to that as the first group and the `*/` as the second group.

Seems fine to me ... but like I said ... maybe there's a pitfall here I don't know about.

**Comment by chrisgrieser** [comment]
just tested it, works fine on my machine. Thanks a lot! :)
